tests/test_notify_suppliers_with_incomplete_applications.py

```diff
@@ -0,0 +1,218 @@
+#                 'supportLevels': 'one off cost', 
+#                 'systemRequirements': ['N/A', 'N/A'], 
+#                 'updatedAt': '2018-05-10T08:22:31.730794Z', 
+#                 'usageNotifications': False, 
+#                 'webChatSupport': 'no', 
+#                 'webInterface': False
+#             }
+#         ]
+#     },
+# ]
+
+# USERS_TEST_CASES = [
+#     {
+#         'users': []
+#     },
+#     {
+#         'users': [
+#             {
+#                 'active': False,
+#                 'emailAddress': 'abc@example.com'
+#             },
+#             {
+#                 'active': True,
+#                 'emailAddress': 'abc@example.com'
+#             },
+#         ]
+#     },
+# ]
+
+MESSAGE_TEST_CASES = (
+    (
+        {
+            'services': []
+        },
+        [
+            {
+                'supplierId': 0,
+                'applicationCompanyDetailsConfirmed': True,
+                'declaration': {'status': 'submitted', 'primaryContactEmail': 'abc@example.com'}
+            },
+            {
+                'supplierId': 1,
+                'applicationCompanyDetailsConfirmed': None,
+                'declaration': {'status': 'submitted', 'primaryContactEmail': 'abc@example.com'}
+            },
+        ],
+        {
+            'users': [
+                {
+                    'active': False,
+                    'emailAddress': 'abc@example.com'
+                },
+                {
+                    'active': True,
+                    'emailAddress': 'abc@example.com'
+                },
+            ]
+        },
+        4,
+        '',
+    ),
+)
+
+
+@pytest.mark.parametrize(
+    'draft_services_case,framework_supplier_case,users_case,expected_mails,expected_message', MESSAGE_TEST_CASES
+)
+@mock.patch('dmscripts.notify_suppliers_with_incomplete_applications.get_api_endpoint_from_stage', autospec=True)
+@mock.patch('dmscripts.notify_suppliers_with_incomplete_applications.get_auth_token', autospec=True)
+@mock.patch('dmscripts.notify_suppliers_with_incomplete_applications.scripts_notify_client', autospec=True)
+@mock.patch('dmscripts.notify_suppliers_with_incomplete_applications.DataAPIClient', autospec=True)
+def test_message_combinations(
+    data_api_client_mock, mail_client_constructor_mock,
+    get_auth_token_constructor_mock, get_api_endpoint_from_stage_constructor_mock,
+    draft_services_case, framework_supplier_case, users_case, expected_mails, expected_message
+):
+    """
+    Test if the correct number of email are sent and with the correct message.
+    """
+    mail_client_mock = mail_client_constructor_mock.return_value = mock.Mock(spec=DMNotifyClient)
+    mail_client_mock.logger = mock.Mock(spec=Logger)
+    data_api_client_mock().find_draft_services.return_value = draft_services_case
+    data_api_client_mock().find_framework_suppliers_iter.return_value = framework_supplier_case
+    data_api_client_mock().find_users.return_value = users_case
```

Comment:
Not essential for this script but we usually assert the `data_api_client_mock` calls as well as the notify client mocks (if we were posting/patching I'd say it was more important that we assert).

Reply:
I will keep it in mind for the future.